service/src/main/resources/config/broad/aou_synthetic/expanded/entity/condition.json

```diff
@@ -201,7 +201,8 @@
               "column" : "concept_id"
             }
           }
-        }
+        },
+        "maxHierarchyDepth" : 20
```

Comment:
For someone configuring a new dataset, how would someone know this value? And what would be the consequences of getting it wrong?

Reply:
You mentioned a script (AoU-CB bash scripts). Could you point to them?

Maybe add this documentation to HierarchyMapping.java.

Instead of setting it per underlay, could we just set it to something like 50 for all underlays? I'm guessing the purpose is to avoid infinite loop... ?

Reply:
Unfortunately, no easy answer to "how would someone know this value" -- basically they have to know enough about the dataset to know how deep the hierarchy is.

If you get it wrong by overshooting, then the hierarchy will still be generated correctly, but the indexing job will take longer. If you get it wrong by undershooting, then the hierarchy won't be generated correctly -- it will be missing levels.

Previously, the max depth was set to 64 for all underlays. That worked fine and is still the default, but lowering the number to something more realistic makes the indexing jobs run faster. This shouldn't ever cause infinite loops, it's just that Apache Beam (doesn't support conditional steps)[https://issues.apache.org/jira/browse/BEAM-106], so we'll always try to compute the max hierarchy depth, even if there are no more parent-child relationships to expand. The only way to reduce the size of the indexing job is to set the max depth lower beforehand.

Reply:
Added the link to the AoU-CB bash scripts to the PR description.

Added a comment to `HierarchyMapping.DEFAULT_MAX_HIERARCHY_DEPTH` about how overshooting is ok, but undershooting is not.

Reply:
It should be possible to give the user feedback on the max depth:
- if they overshot, then compute the max hierarchy depth and report it out.
- if they undershot, then I'm not sure we can tell them what the depth is exactly, but we can tell them it's bigger than the value they chose/defaulted to.

This could be a good future improvement.

Reply:
Yeah, that would be a good improvement. The silent failure is the really bad problem. The documentation improvements are good for the short term at least.

Reply:
Added JIRA ticket for this https://precisionmedicineinitiative.atlassian.net/browse/TNG-94